### ISSUE TITLE
Unable to override ValidationTagLib in Grails 3.1.12

### ISSUE THREAD

**Issue by kalpanaDasireddy** [grails/grails-core#10311]
I'm trying to override the default g:message tablib to implement custom behaviour by creating a custom taglib that extends ValidationTagLib.

I see the following info message in server startup logs and the g:message from ValidationTagLib is being called instead of from my taglib.

org.grails.core.artefact.TagLibArtefactHandler - There are conflicting tags: com.xms.taglib.XgMessageTagLib.g:message vs. org.grails.plugins.web.taglib.ValidationTagLib.g:message. The former will take precedence.

The same implementation worked in grails 2.2.3. Is there any way to override taglibs in grails 3.1.12

I'm doing it like this.
class XgMessageTagLib extends ValidationTagLib 
{
         Closure message = { attrs ->
                  .......................
          }
}

**Comment by jeffbrown** [comment]
See the following:

https://github.com/jeffbrown/issue10311/blob/58c23546b294fd3d79e55863a77ed1f61accc813/grails-app/taglib/demo/CustomTagLib.groovy

https://github.com/jeffbrown/issue10311/blob/58c23546b294fd3d79e55863a77ed1f61accc813/grails-app/views/index.gsp#L58-L60

**Comment by jeffbrown** [comment]
Does the tag linked above work for you?

**Comment by kalpanaDasireddy** [comment]
Yes. It works. I have other taglibs which works perfectly fine. Its only when trying to override the default behaviour of g:message tag lib using my own taglib, since both the taglibs have same namespace, the version if taglib in ValidationTagLib class takes precedence over my custom taglib.


I resolved this by not extending the taglib from ValidationTagLib. This way my taglib is loaded before ValidationTagLib.

**Comment by jeffbrown** [comment]
Great.  That is also what I did in the sample project linked above.

**Comment by jeffbrown** [comment]
> I resolved this by not extending the taglib from ValidationTagLib. This way my taglib is loaded before ValidationTagLib.

The order that the classes are loaded by the JVM isn't really the issue.

**Comment by jeffbrown** [comment]
@kalpanaDasireddy I would like to investigate a bit.  Can you provide a link to a sample project which demonstrates the problem?

Thanks.

**Comment by kalpanaDasireddy** [comment]
@jeffbrown Sorry for the delay in response. I have uploaded a sample project to https://github.com/kalpanaDasireddy/MessageTagLib
In this project, taglib is in the webapp and so 
On the index.gsp the message resolved is "Resolved from CustomMessageTagLib" correctly.

Where as in project 
https://github.com/kalpanaDasireddy/MessageTagLibInPlugin
where the CustomMessageTagLib is in plugin, Message is resolved to Resolved from ValidationTagLib and CustomMessageTagLib  is not called.

**Comment by crudolf** [comment]
I have the same issue. I don't understand where the implementation of message is coming from: I changed the behaviour of the message resolver but only calling message from a taglib takes the old implementation. I deleted all build folders, including .grails, .gradle, out. I called "grails clean" many times.

The only place where the old implementation is called is from calling `message(code: "")` from inside a taglib. If I call `g.message(code: "")` everything works fine.
